satpy/readers/hsaf_nc.py

```diff
@@ -0,0 +1,157 @@
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Reader for H SAF blended precipitation products in NetCDF format.
+
+These products provide precipitation estimates derived from a blending of
+geostationary (GEO) infrared and low-Earth-orbit (LEO) microwave observations
+using the H SAF "Rapid Update" algorithm.
+
+Currently, this reader supports the following products:
+  * **H60/H60B** – Blended GEO/IR and LEO/MW instantaneous precipitation over
+    the full Meteosat (0°) disk.
+  * **H63** – Blended GEO/IR and LEO/MW instantaneous precipitation over the
+    Meteosat IODC (Indian Ocean) disk.
+  * **H90** – Accumulated Precipitation at ground by blended MW and IR IODC
+    over the Meteosat IODC (Indian Ocean) disk.
+
+Notes:
+  * Externally compressed files (.gz) and uncompressed files are handled
+    transparently via Satpys ``generic_open``, without creating temporary
+    decompressed files on disk.
+  * The reader uses area definitions specified in the associated YAML
+    configuration file to determine projection, coverage, and resolution.
+  * Provides access to the ``rain_rate`` and ``q_ind`` datasets (including
+    accumulated variants where available).
+"""
+
+import datetime as dt
+import logging
+
+import xarray as xr
+
+from satpy.area import get_area_def
+from satpy.readers.core.file_handlers import BaseFileHandler
+from satpy.readers.core.utils import generic_open
+from satpy.utils import get_legacy_chunk_size
+
+LOG = logging.getLogger(__name__)
+
+platform_translate = {"MSG1": "Meteosat-8",
+                      "MSG2": "Meteosat-9",
+                      "MSG3": "Meteosat-10",
+                      "MSG4": "Meteosat-11",
+                      }
+
+CHUNK_SIZE = get_legacy_chunk_size()
+
+class HSAFNCFileHandler(BaseFileHandler):
+    """Handle H SAF NetCDF files, with optional .gz external compression.
+
+    This file handler handles H SAF Instantaneous Rain Rate products which contain:
+    - rr: rain rate in mm/h (instantaneous) or mm (accumulated)
+    - qind: quality index in percent
+
+    The data is on a geostationary projection grid.
+    This file handler is tested with H SAF h60/h60b, h63 & h90 data in NetCDF
+    format.
+    """
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Create a wrapper to opens the nc file and store the nc data."""
+        super().__init__(filename, filename_info, filetype_info)
+        self._area_name = None
+        self._lon_0 = None
+        with generic_open(filename, mode="rb", compression="infer") as fp:
+            self._nc_data = xr.open_dataset(fp, chunks="auto", engine="h5netcdf").compute()
```

Comment:
I believe the `compute()` here is breaking the lazy reading of the data... If you remove it, does the reader still work, and we get dask arrays for the main variables? that should be the case..